mesonbuild/compilers/rust.py

```diff
@@ -221,6 +222,23 @@ def get_assert_args(self, disable: bool, env: 'Environment') -> T.List[str]:
         action = "no" if disable else "yes"
         return ['-C', f'debug-assertions={action}', '-C', 'overflow-checks=no']
 
+    def has_arguments(self, args: T.List[str], env: 'Environment', code: str,
+                      mode: CompileCheckMode) -> T.Tuple[bool, bool]:
+        return self.compiles(code, env, extra_args=args, mode=mode)
```

Comment:
"has_arguments" isn't really part of the compiler API, it's just a convenience method for sharing. In C/C++ it's convenient to have a common helper and then compile different code for compile vs. link, since the compile-only version just defines some variables without linking while the link check has to create a `main()` function. You're not doing this here, which implies maybe you should just inline the code?